lib/mongoid/full_text_search.rb

```diff
@@ -289,8 +289,8 @@ def remove_from_ngram_index
       end
     end
 
-    def update_ngram_index
-      all.each(&:update_ngram_index)
+    def update_ngram_index(no_timeout: false)
```

Comment:
While the mongoid method is `no_timeout`, it's merely a facility to pass a timeout value. I think we should flip this one to be `timeout` and afford for future options as well.

```ruby
def update_ngram_index(options = { timeout: true })
   items = all
   items = items.no_timeout if options.key?(:timeout) && !options[:timeout]
   items.each(&:update_ngram_index)
end
```

What do you think?

Reply:
Users will write `Artwork.update_ngram_index(timeout: false)` which feels better.

Reply:
I totally agree. I think this feels more natural and doesn't just expose a mongoid method as an optional boolean